File: hacker_rank/week1/plus_minus.py
size = int(input())
arr = list(map(int, input().split()))


pos_count = 0
neg_count = 0
zero_count = 0


for num in arr:
    if num > 0:
        pos_count += 1
    elif num == 0:
        zero_count += 1
    elif num < 0:
        neg_count += 1


p_ratio = pos_count / size
n_ratio = neg_count / size
z_ratio = zero_count / size


print(f"{p_ratio:.6f}")
print(f"{n_ratio:.6f}")
print(f"{z_ratio:.6f}")

File: hacker_rank/week1/plus_minus.c
#include <stdio.h>

int main() {
    int size;
    scanf("%d", &size);

    int arr[size];
    for(int i = 0; i < size; i++) {
        scanf("%d", &arr[i]);
    }

    int positive = 0, negative = 0, zero = 0;
    for(int i = 0; i < size; i++) {
        if(arr[i] > 0) {
            positive++;
        } else if(arr[i] == 0) {
            zero++;
        } else {
            negative++;
        }
    }

    double p_ratio = (double)positive / size;
    double n_ratio = (double)negative / size;
    double z_ratio = (double)zero / size;

    printf("%.6lf\n", p_ratio);
    printf("%.6lf\n", n_ratio);
    printf("%.6lf\n", z_ratio);

    return 0;
}
